src/repositories/pog-profile.repository.ts

```diff
@@ -182,6 +191,20 @@ export class PogProfileRepository extends DefaultCrudRepository<
     return undefined;
   }
 
+  async addEmailToProfile(userId: string, email: string) {
+    const emailHash = this.hashEmail(email);
+    const emailQuerySnapshot = await this.db
+      .collection('pogprofiles')
+      .where('emailHash', '==', emailHash)
+      .get();
+    if (emailQuerySnapshot?.docs?.length) {
+      throw new Error('Email already exists in another profile.');
+    }
+
+    await this.db.doc(`pogprofiles/${userId}`).update({email, emailHash});
+    return {email, emailHash};
```

Comment:
switching this to a returning function so that the caller function can forge a profile with the emailHash